backend/tracim_backend/command/cleanup.py

```diff
@@ -166,36 +168,44 @@ def take_app_action(self, parsed_args: argparse.Namespace, app_context: AppEnvir
-                                workspace_id
-                            )
+            app_lib = ApplicationApi(app_list=app_list)
+            if app_lib.exist("agenda"):
+                # INFO - G.M - 2019-12-13 - cleanup agenda at end of process
+                if deleted_workspace_ids:
+                    deleted_workspace_ids_str = [
+                        '"{}"'.format(workspace_id) for workspace_id in deleted_workspace_ids
+                    ]
+                    print(
+                        "delete agenda of workspaces {}".format(
+                            ", ".join(deleted_workspace_ids_str)
                         )
-                        print(traceback.format_exc())
+                    )
+                    for workspace_id in deleted_workspace_ids:
+                        try:
+                            cleanup_lib.delete_workspace_agenda(workspace_id)
+                        except AgendaNotFoundError:
+                            print(
+                                'Warning: Cannot delete agenda for workspace "{}", agenda not found. Agenda path may be incorrect or agenda not created'.format(
+                                    workspace_id
+                                )
+                            )
+                            print(traceback.format_exc())
 
-            if deleted_user_ids:
-                deleted_user_ids_str = ['"{}"'.format(user_id) for user_id in deleted_user_ids]
-                print("delete agenda of users {}".format(", ".join(deleted_user_ids_str)))
-                for user_id in deleted_user_ids:
-                    try:
-                        cleanup_lib.delete_user_agenda(user_id)
-                    except AgendaNotFoundError:
-                        print(
-                            'Warning: Cannot delete agenda for user "{}", agenda not found. Agenda path may be incorrect or agenda not created'.format(
-                                user_id
+                if deleted_user_ids:
+                    deleted_user_ids_str = ['"{}"'.format(user_id) for user_id in deleted_user_ids]
+                    print("delete agenda of users {}".format(", ".join(deleted_user_ids_str)))
+                    for user_id in deleted_user_ids:
+                        try:
+                            cleanup_lib.delete_user_agenda(user_id)
+                        except AgendaNotFoundError:
+                            print(
+                                'Warning: Cannot delete agenda for user "{}", agenda not found. Agenda path may be incorrect or agenda not created'.format(
+                                    user_id
+                                )
                             )
-                        )
-                        print(traceback.format_exc())
+                            print(traceback.format_exc())
+            else:
+                print("Warning ! Agenda app not enabled, agenda will not be deleted.")
```

Comment:
We do not have anymore config of app if these app are not loaded, so we can not anymore expect caldav calendar to be delete in agenda app is not loaded. Not sure how and if we do need to fix this.

Reply:
@lebouquetin is this new behavior ok ?

Reply:
Is it possible to "force" activate agenda application when delete users ?

Reply:
Yes, we can as we load app a disabled_app.
 we can do something like: 

```python
app_lib = ApplicationApi(app_list=app_list, show_disabled=True)
if app_lib.exist(AGENDA__APP_SLUG):
    agenda_app = app_lib.get(AGENDA__APP_SLUG)
    if not agenda_app.is_active:
        agenda_app.load_config(self._app_config)
```

Not sure if it make sense to do agenda_app.check_config() too.

Reply:
if we do control config :
- we will get explicit error if config is uncorrect
- we will support only fully correct agenda config

if we do not control config:
- we will get implicit error if config is uncorrect.
- we will support all config with correct agenda dir parameter (if other parameter related to caldav are uncorrect, we do not check them).

Reply:
@inkhey could we have a chat about this?

Reply:
After discussion with @lebouquetin, we decide to not force agenda deletion if agenda app is no activated.